test/TestUtilities/Test.Utility/TestServer/TcpListenerServer.cs

```diff
@@ -53,6 +71,50 @@ public async Task<T> ExecuteAsync<T>(Func<string, Task<T>> action)
                 CancellationToken.None);
         }
 
+        public TcpListenerServer()
+        { }
+
+        private static X509Certificate2 GenerateSelfSignedCertificate()
+        {
+            using (var rsa = RSA.Create(2048))
+            {
+                var request = new CertificateRequest("cn=test", rsa, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+                var start = DateTime.UtcNow;
+                var end = DateTime.UtcNow.AddYears(1);
+                var cert = request.CreateSelfSigned(start, end);
+                var certBytes = cert.Export(X509ContentType.Pfx, "password");
+
+                return new X509Certificate2(certBytes, "password", X509KeyStorageFlags.Exportable);
+            }
+        }
+
+        private async Task StartInvalidTlsCertificateServer(TcpListener tcpListener, CancellationToken token)
+        {
+            while (!token.IsCancellationRequested)
+            {
+                using (var client = await Task.Run(tcpListener.AcceptTcpClientAsync, token))
+                using (var sslStream = new SslStream(client.GetStream(), false))
+                {
+                    sslStream.AuthenticateAsServer(_tlsCertificate, clientCertificateRequired: false, SslProtocols.Tls12, checkCertificateRevocation: true);
```

Comment:
Can we just remove `SslProtocols.Tls12` argument as discussed in `SelfSignedCertificateMockServer.cs` implementation?